pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -93,10 +107,30 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
-		err = client.SetToken(path.Join(instance.Spec.TenantRef.Name, instance.Name, "steward"), token, reqLogger)
+		err = vault.SetToken(secretPath, token)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		vault.SetDeletePaths([]string{secretPath})
+
+		deleted, err := helpers.HandleDeletion(instance, vault, finalizerName, r.client)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		if deleted {
+			err = r.removeClusterFileFromTenant(instance.GetName(), repoName, reqLogger)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+	}
+
+	deleted, err := helpers.HandleDeletion(instance, nil, finalizerName, r.client)
```

Comment:
Seems kinda weird to call `HandleDeletion` again here. Maybe move it out of the `if` and only call it once?

Reply:
Hmmm... if I create a nil vault client before the if this could work.